packages/server/lib/controllers/sync.controller.ts

```diff
@@ -411,11 +415,13 @@ class SyncController {
 
             if (isOk(actionResponse)) {
                 res.send(actionResponse.res);
+                span.finish();
 
                 return;
             } else {
                 span.setTag('nango.error', actionResponse.err);
                 errorManager.errResFromNangoErr(res, actionResponse.err);
+                span.finish();
```

Comment:
let's add span.finish in the catch after setting the error tag as well and we also don't need the finally anymore then

Reply:
Updated

Reply:
why the finally didn't cover all of those changes?

Reply:
We were chatting about this and @TBonnin theorized that because the controller is using a `res.send` perhaps the span wasn't getting closed because the request itself was ended.